server/datastore/mysql/labels.go

```diff
@@ -460,16 +460,33 @@ func (ds *Datastore) ListLabelsForHost(ctx context.Context, hid uint) ([]*fleet.
 // ListHostsInLabel returns a list of fleet.Host that are associated
 // with fleet.Label referened by Label ID
 func (ds *Datastore) ListHostsInLabel(ctx context.Context, filter fleet.TeamFilter, lid uint, opt fleet.HostListOptions) ([]*fleet.Host, error) {
-	query := `
-			SELECT
-				h.*,
-				COALESCE(hst.seen_time, h.created_at) as seen_time,
-				(SELECT name FROM teams t WHERE t.id = h.team_id) AS team_name
-			FROM label_membership lm
-			JOIN hosts h ON (lm.host_id = h.id)
-			LEFT JOIN host_seen_times hst ON (h.id=hst.host_id)
-			WHERE lm.label_id = ?
+	query := `SELECT
+		h.*,
+		COALESCE(hst.seen_time, h.created_at) as seen_time,
+		(SELECT name FROM teams t WHERE t.id = h.team_id) AS team_name
+	`
+
+	failingPoliciesSelect := `,
+		coalesce(failing_policies.count, 0) as failing_policies_count,
+		coalesce(failing_policies.count, 0) as total_issues_count
 	`
+	if opt.DisableFailingPolicies {
+		failingPoliciesSelect = ""
+	}
+	query += failingPoliciesSelect
+
+	query += `FROM label_membership lm
+		JOIN hosts h ON (lm.host_id = h.id)
+		LEFT JOIN host_seen_times hst ON (h.id=hst.host_id)`
+
+	failingPoliciesJoin := `LEFT JOIN (
+		SELECT host_id, count(*) as count FROM policy_membership WHERE passes = 0
+		GROUP BY host_id
+	) as failing_policies ON (h.id=failing_policies.host_id)`
+	if opt.DisableFailingPolicies {
+		failingPoliciesJoin = ""
+	}
+	query += failingPoliciesJoin
```

Comment:
I think this is a matter of personal preference. But I would do this more like this:

```
	queryFmt := `
			SELECT
				h.*,
				COALESCE(hst.seen_time, h.created_at) as seen_time,
				(SELECT name FROM teams t WHERE t.id = h.team_id) AS team_name
				%s
			FROM label_membership lm
			JOIN hosts h ON (lm.host_id = h.id)
			LEFT JOIN host_seen_times hst ON (h.id=hst.host_id)
			%s
	`

	failingPoliciesSelect := `,
		coalesce(failing_policies.count, 0) as failing_policies_count,
		coalesce(failing_policies.count, 0) as total_issues_count
	`
	if opt.DisableFailingPolicies {
		failingPoliciesSelect = ""
	}
	failingPoliciesJoin := `LEFT JOIN (
		SELECT host_id, count(*) as count FROM policy_membership WHERE passes = 0
		GROUP BY host_id
	) as failing_policies ON (h.id=failing_policies.host_id)`
	if opt.DisableFailingPolicies {
		failingPoliciesJoin = ""
	}

	query := fmt.Sprintf(queryFmt, failingPoliciesSelect, failingPoliciesJoin)
```

The goal would be to see the overall shape of the query and then the things that may vary. Neither way is super readable, sadly, though.